Library/Homebrew/formula_creator.rb

```diff
@@ -100,6 +100,7 @@ def write_formula!
     sig { params(name: String).returns(String) }
     def latest_versioned_formula(name)
       name_prefix = "#{name}@"
+      CoreTap.instance.clear_cache
```

Comment:
Why is it needed here specifically?

Reply:
Good question. I may have misinterpreted @cho-m's comment: https://github.com/Homebrew/brew/pull/19244#issuecomment-2646364313